packages/sdk/src/encryption/EncryptionUtil.ts

```diff
@@ -1,28 +1,146 @@
 import crypto, { CipherKey } from 'crypto'
+import { ml_kem1024 } from '@noble/post-quantum/ml-kem'
+import { randomBytes } from '@noble/post-quantum/utils'
 import { StreamMessageAESEncrypted } from '../protocol/StreamMessage'
 import { StreamrClientError } from '../StreamrClientError'
 import { GroupKey } from './GroupKey'
+import { AsymmetricEncryptionType } from '@streamr/trackerless-network'
+import { binaryToUtf8 } from '@streamr/utils'
+import { getSubtle } from '../utils/crossPlatformCrypto'
 
 export const INITIALIZATION_VECTOR_LENGTH = 16
 
+const INFO = Buffer.from('streamr-key-exchange')
+const KEM_CIPHER_LENGTH_BYTES = 1568
+const KDF_SALT_LENGTH_BYTES = 64
+
 // eslint-disable-next-line @typescript-eslint/no-extraneous-class
 export class EncryptionUtil {
-    private static validateRSAPublicKey(publicKey: crypto.KeyLike): void | never {
-        const keyString = typeof publicKey === 'string' ? publicKey : publicKey.toString('utf8')
-        if (typeof keyString !== 'string' || !keyString.startsWith('-----BEGIN PUBLIC KEY-----')
+    /**
+     * Generic utility functions
+     */
+    static async encryptForPublicKey(plaintext: Uint8Array, publicKey: Uint8Array, type: AsymmetricEncryptionType): Promise<Buffer> {
+        if (type === AsymmetricEncryptionType.ML_KEM) {
+            return this.encryptWithMLKEMPublicKey(plaintext, publicKey)
+        }
+        if (type === AsymmetricEncryptionType.RSA) {
+            return this.encryptWithRSAPublicKey(plaintext, publicKey)
+        }
+        throw new Error(`Unexpected GroupKeyEncryptionType: ${type}`)
```

Comment:
Maybe could reword the error message as these are generic utility functions (can be used to encrypt/decrypt non-groupkey data in the future)

```suggestion
        throw new Error(`Unexpected encryption type: ${type}`)
```